app/com/arpnetworking/rollups/QueueActor.java

```diff
@@ -0,0 +1,255 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.rollups;
+
+import akka.actor.AbstractActor;
+import akka.actor.Props;
+import akka.japi.pf.ReceiveBuilder;
+import com.arpnetworking.logback.annotations.Loggable;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.ArrayDeque;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Queue;
+
+/**
+ * An actor with a bounded queue.
+ *
+ * @param <T>
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public class QueueActor<T extends Serializable> extends AbstractActor {
+    /**
+     * Creates a {@link Props} for this actor.
+     *
+     * @param maxSize The maximum size for the queue (if any).
+     * @return A new Props.
+     */
+    public static Props props(final Optional<Long> maxSize) {
+        return Props.create(QueueActor.class, maxSize);
+    }
+
+    @Override
+    public Receive createReceive() {
+        return new ReceiveBuilder()
+                .match(Add.class, message -> {
+                    final T item;
+                    try {
+                        // TODO(spencerpearson): I'm not sure what the Right Way is to address this. Maybe there isn't one.
+                        @SuppressWarnings("unchecked") final T intermediate = (T) message.getItem();
+                        item = intermediate;
+                    } catch (final ClassCastException err) {
+                        getSender().tell(new AddRejected<>(message.getItem()), getSelf());
+                        return;
+                    }
+                    if (_maxSize.isPresent() && _queue.size() < _maxSize.get()) {
+                        _queue.add(item);
+                        getSender().tell(new AddAccepted<>(item), getSelf());
+                    } else {
+                        getSender().tell(new AddRejected<>(item), getSelf());
+                    }
+                })
+                .match(Poll.class, request -> {
+                    getSender().tell(
+                            Optional.ofNullable((Object) _queue.poll())
+                                    .orElse(QueueEmpty.getInstance()),
+                            getSelf()
+                    );
+                })
+                .build();
+    }
+
+    /**
+     * {@link QueueActor} actor constructor.
+     *
+     * @param maxSize the maximum size of the queue
+     */
+    public QueueActor(final Optional<Long> maxSize) {
+        _maxSize = maxSize;
+    }
+
+    private final Optional<Long> _maxSize;
```

Comment:
I see Ville has infected you with putting your members in the wrong place. ;)

Reply:
Hey, I was just copying _your_ `RollupGenerator`! :P What's the right place?

Reply:
I think Ville relaxed this but the Java defacto standard is to put members at the top of the class followed by constructor and then public and private methods.

Reply:
Sounds good! Reorganized.